src/sagas/session.js

```diff
@@ -63,6 +63,8 @@ export function* listBuckets(getState, action) {
     // Fetch server information
     const serverInfo = yield call([client, client.fetchServerInfo]);
     // We got a valid response; officially declare current user authenticated
+    // XXX: authenticated here means that we have setup the client, not
+    // that then credentials are valid :/
```

Comment:
s/then/the 

I agree that this seems wrong, we should probably mark a user as officially authenticated only when the server has acknowledged the provided creds. Maybe in another issue.